src/i2c_master_setup.c

```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+#include <zephyr.h>
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+
+/*Status LED DT reference*/
+#define LED0_NODE DT_ALIAS(led2)
```

Comment:
Since this is an I2C file, the LED's definitions would be better if defined their own file and included that file here so that there is more cleaner implementation.